### ISSUE TITLE
 dart run drift_dev analyze  Unhandled exception: Null check operator used on a null value

### ISSUE THREAD

**Issue by MahmoodAliL** [simolus3/moor#2005]
The bugs happen when the SQL formatter adds a space between $ and the variable name, It takes me a lots of time to identify the issue , please consider add a clearly message to allow us to fix the bugs faster.

```
SELECT
    person.id,
    person.name
FROM
    person
ORDER BY
    $ order;
```

TO 
```

SELECT
    person.id,
    person.name
FROM
    person
ORDER BY
    $order


```
Logs: 

 ```
dart run drift_dev analyze                  
Unhandled exception:
Null check operator used on a null value
#0      Scanner._scanToken (package:sqlparser/src/reader/tokenizer/scanner.dart:168:40)
#1      Scanner.scanTokens (package:sqlparser/src/reader/tokenizer/scanner.dart:39:7)
#2      SqlEngine.tokenize (package:sqlparser/src/engine/sql_engine.dart:102:28)
#3      SqlEngine.parseDriftFile (package:sqlparser/src/engine/sql_engine.dart:123:20)
#4      MoorParser.parseAndAnalyze (package:drift_dev/src/analyzer/drift/parser.dart:18:27)
#5      ParseMoorStep.parseFile (package:drift_dev/src/analyzer/runner/steps/parse_moor.dart:13:19)
<asynchronous suspension>
#6      Task._parse (package:drift_dev/src/analyzer/runner/task.dart:104:20)
<asynchronous suspension>
#7      Task.runTask (package:drift_dev/src/analyzer/runner/task.dart:49:20)
<asynchronous suspension>
#8      MoorDriver.waitFileParsed (package:drift_dev/src/backends/common/driver.dart:131:9)
<asynchronous suspension>
#9      AnalyzeCommand.run (package:drift_dev/src/cli/commands/analyze.dart:27:23)
<asynchronous suspension>
#10     CommandRunner.runCommand (package:args/command_runner.dart:209:13)
<asynchronous suspension>
#11     MoorCli.run (package:drift_dev/src/cli/cli.dart:65:5)
<asynchronous suspension>
#12     run (package:drift_dev/src/cli/cli.dart:19:12)
<asynchronous suspension>
```

**Comment by simolus3** [comment]
Thanks for the report! I've fixed this in 70b18550d9c2dbd6b09574005c543fffebb0f5d3 and I'll release this fix sometime this weekend.

**Comment by MahmoodAliL** [comment]
Thank you
